First of all it could be interesting to study the variance of expression $\var{\mathrm{counts}}$ versus 
the average $\avg{\mathrm{counts}}$ across tissues.
%%all genes
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/allgenes/varmean_loglog.png}
    \caption{Variance versus average. In red the Poisson-like scaling, in green the Taylor-like scaling. All genes are considered}
    \label{fig:scalelaws/gtex/allgenes/varmean_loglog_density}
\end{figure}
In figure~\ref{fig:scalelaws/gtex/varmean_loglog_density} the scatter plot of variance versus mean reveal some interesting facts.
First of all it is evident that data have a double scaling behaviour: when the mean is small ($\lesssim 1$) data have a Poisson-like scaling ($\var{\mathrm{counts}} \sim \avg{\mathrm{counts}}$), at higher means instead data have a quadratic scaling ($\var{\mathrm{counts}} \sim \avg{\mathrm{counts}}^2$) known in ecology as Taylor's law~\cite{Eisler2008}. This means that at low averages data behaviour is just due to the sampling experimental process, on the contrary the Taylor's law reveals the non trivial distribution across samples of the gene expression.
Another interesting fact is that looking at the density of points (colours in figure~\ref{fig:scalelaws/gtex/allgenes/varmean_loglog_density}) are evident two clouds of points, one at low averages, one at high averages. These correspond to coding and non coding genes, remembering section~\ref{sec:universallaws} these two kind of genes have a different behaviour: coding genes are highly expressed in the majority of the samples, non coding ones are less expressed (and so less sampled) in few samples. 

A similar analysis, common in literature, is the analysis of the coefficient of variation squared $CV^2=\frac{\var{\mathrm{counts}}}{\avg{\mathrm{counts}}^2}$ represented in figure~\ref{fig:scalelaws/gtex/allgenes/cvmean_loglog}.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/allgenes/cvmean_loglog.png}
    \caption{Coefficient of variation squared versus average. In red the Poisson-like scaling, in green the Taylor-like scaling}
    \label{fig:scalelaws/gtex/allgenes/cvmean_loglog}
\end{figure}
The behaviour is complementary to the above discussed double scaling and is quite common in literature looking at single cell RNA sequencing data~\cite{Islam2013}. Even looking at $CV^2$ it is evident the presence of the coding and non-coding clouds of points. The non coding genes are on the Poisson-like scaling, $\var{\mathrm{counts}} \sim \avg{\mathrm{counts}}$ so $CV^2=\frac{\var{\mathrm{counts}}}{\avg{\mathrm{counts}}^2}\sim\frac{1}{\avg{\mathrm{counts}}}$, otherwise the protein coding genes are on the Taylor-like curve $CV^2=\frac{\var{\mathrm{counts}}}{\avg{\mathrm{counts}}^2}\sim 1$.

\paragraph{Protein coding genes} can be isolated and considered on their own. The same analysis confirms that the cloud of genes' points on the Taylor-like scaling are effective the protein coding genes.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/varmean_loglog_density.png}
    \caption{Variance versus average. In red the Poisson-like scaling, in green the Taylor-like scaling. Only protein coding genes are considered}
    \label{fig:scalelaws/gtex/varmean_loglog_density}
\end{figure}
Following the sampling model of~\cite{Mazzolini2018} sum up in section~\ref{sec:nullmodel} the averages and variances can be estimated on null matrices. In figure~\ref{fig:scalelaws/gtex/varmean_3sigma} the comparison between real genes and sampling ones. The sampling has got a double scaling as well; this is quite interesting, it means that the global scaling is due to the Zipf distribution and the sizes distribution themselves, they are identical in data and sampling by definition.
Moreover the sampling points draw a lower bound of the data, this encodes the information that the data are more variable (have higher variance) than just sampling, so there must be some biological information hidden that causes this over variable behaviour.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/varmean_3sigma.png}
    \caption{Variance versus average. In \textcolor{pythonred}{red} the Poisson-like scaling, in \textcolor{pythongreen}{green} the Taylor-like scaling. In \textcolor{pythonorange}{orange} the sampling components. Only protein coding genes are considered}
    \label{fig:scalelaws/gtex/varmean_3sigma}
\end{figure}

Again it is possible to analyse the $CV^2$, at this time considering only protein coding genes. Figure~\ref{fig:scalelaws/gtex/allgenes/cvmean_loglog} confirms that the cloud of points near the Taylor-like scaling are the protein coding genes and a double scaling is seen once again.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/cvmean_loglog_density.png}
    \caption{Variance versus occurrence}
    \label{fig:scalelaws/gtex/cvmean_loglog}
\end{figure}

In figure~\ref{fig:scalelaws/gtex/cvmean_loglog_sampling} the same plot compared to the sampling points. The double scaling is evident also for the sampling points. Note that $CV^2$ has got a lower bound at $0$ which corresponds to the less variable case if all expression are identical in all samples ($\var{\mathrm{counts}}$) and an upper bound at $N-1$ with $N$ the number of realisations and corresponds to the most variable case where a component  express in only one realisation and is $0$ elsewhere.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/scalelaws/gtex/cvmean_loglog_sampling.png}
    \caption{Caption}
    \label{fig:scalelaws/gtex/cvmean_loglog_sampling}
\end{figure}

Finally the data have a double scaling when looking at their global variance across realisations, a Poisson-like where the sampling experimental process is more important and a Taylor-like where the complexity of the data emerges.
Non coding genes have got low expression and are rare, protein coding genes, otherwise, express a lot and everywhere and carry more information following a double scaling. All genes are more variable than a sampling null model and this is the evidence that something interesting is hidden behind the data.
