\chapter{Conclusions}\label{ch:conclusions}
Finally this work demonstrates that RNA-Sequencing datasets can be analyses from a component systems point of view.

Gene expression data show typical trends well-known, for example, in linguistics, moreover some interesting biological signatures were found. RNA-sequencing datasets have a great core of protein-coding genes that express everywhere, this is evident looking at $U$s or at Heaps' law. The presence of a power law distribution in the ranked abundances, the so called Zipf's law is observed and characterize the distribution of genes expression data.

In the first part of this work a dataset (GTEx) containing samples from healthy tissues was analysed. One of the most interesting evidence was the presence of many different Zipf's law if one considers each tissue independently. Very similar results were obtained considering TCGA, a dataset containing thousands of samples of cancer tissues.

The power law distributions and the similarities with what was found in linguistics encouraged to explore the possibility of using a topic model approach to reveal the hidden structure of these datasets. This approach, originally developed to classify text, was useful both to find clusters of samples that share some properties and to find the relation between genes and samples.

Many goals were achieved during the topic modelling analysis. The pipeline begins filtering the data and selecting useful genes, then this network is processed with hierarchic stochastic block model algorithm, then clusters are analysed and enrichment tests are performed; in the meanwhile, an objective and a well-defined score is estimated. All the analysis confirmed that this approach is successful. Three different datasets were analysed and, in every case, the model performed well. What was found is that clusters contain samples that share some property, in particular, the tissue they are related; enrichment tests found tissue related terms in the topics along with Gene Ontology terms. The relation between genes and samples is non-trivial and revealed a complex structure in gene expression data. Nevertheless, this structure is sufficient to discriminate between tissues.

In conclusion, topic model reveal itself as a useful approach to find the hidden structure of gene expression data. The prior analysis to select highly variables genes make it possible to run the algorithm faster without losing necessary information.

During this work the foundations have been laid for other analysis; for example, it should be interesting to study the variability of gene expression between tissues and between individual, ideally genes that change their behaviour inter different tissues and not intra the same tissue are more likely tissue specific. Trying to remove the sampling effect from the data could be another interesting analysis, in particular reproducing~\cite{Grilli} on RNA-sequencing data could lead to the removal of sampling effects, here this was done just considering the sampling as a $CV^2$ boundary. Applying the model to single cell RNA-sequencing data, maybe from other kinds of animal, will present new challenges not present in bulk RNA-sequencing datasets considered here. Reproduce mouse data from~\cite{Scialdone2016} could be an interesting starting point. Maybe it worth to run the model on data where just sampling is present and verify if there is some bias on the model due to the presence of the sampling.

The main future development of this work is indeed to run the model to a specific cancer tissue and find cancer subtypes, for instance, the Breast or Colon one. Obtaining a hierarchy that at some level is able to identify cancer sub-types would be an ideal and great goal. 