\chapter{Introduction}\label{ch:intro}
In recent years the study of complex systems is becoming more interesting 
especially because some different systems can share interesting and fundamental 
properties. Network theory has proven to be a useful proxy to model and represent such complex 
systems.

This work wants to study and find universal statistical laws in different kind of biological systems.
If one finds that two different systems share some important laws and data structure, therefore it is possible to use tool from different fields 
to study and gain information about each others. 
In particular two datasets containing information about some human healthy and 
diseased tissues will be analysed. This data come from biological experiments of 
RNA sequencing.

The ultimate goal of this work would be to study, develop and build machine learning's methods able to 
discriminate healthy and diseased tissues. Once diseased tissue are found, the 
next goal is to separate cancer types and ultimately sub-types, which is not 
always easy clinically.

The methods to gain this goal are derived firstly from linguistics, in 
particular a topic model approach will be widely described.

In chapter~\ref{ch:data} I will describe the datasets used and introduce some 
basic biological properties of these datasets. In particular I'll use two datasets of gene expression data from 
diseased tissues and healthy tissues.

In chapter~\ref{ch:structure} I will describe the basics of component systems 
and give some basic mathematical definitions of quantities useful in general. 
This chapter refers in particular to the so called component systems.

In chapter~\ref{ch:scalinglaws} I will represent the gene expression from one sample from TCGA 
with respect to the genes' rank, one can easily obtain a Zipf's law. This law is 
well-known and in-depth studied in linguistics, demonstrating that different sources of data (genomic and linguistics) 
can share some statistical properties.

Demonstrated that linguistics and biological data share some laws in section~\ref{ch:topicmodelling} 
I will use topic modelling to perform network analysis on datasets.
Using topic modelling one would find the inner structure of the samples.
One would find clusters such that all samples in a cluster share the tissue and tumour type.
As far as a document can contain a mixture of similar topics a single tumour can be very heterogeneous.

In chapter~\ref{ch:conclusions} I will discuss the results and the future 
developments

Many methods of the pipeline was written in C\texttt{++} using  openMP and Boost~\cite{siek2002boost} are available at~\url{https://github.com/fvalle1/tacos}.
During this work I used different python libraries such as 
pandas~\cite{mckinney2010data}, scipy~\cite{jones2014scipy}, numpy~\cite{oliphant2006guide} and 
matplotlib~\cite{hunter2007matplotlib}. Some analysis required tensorflow~\cite{tensorflow2015-whitepaper} 
and pySpark~\cite{Zaharia:2016:ASU:3013530.2934664}.
The topic modelling stochastic block modelling minimization functions are implemented in the graph-tool library~\cite{peixoto_graph-tool_2014}.
The full work repository is on Github at~\url{http://github.com/fvalle1/master_thesis}.