\chapter*{Introduction}\label{ch:intro}
\addcontentsline{toc}{chapter}{Introduction}
In recent years the study of complex systems is becoming more interesting especially when some different systems that share interesting and fundamental properties are found. Network theory has been proven to be a useful proxy to model and represent such complex systems.

This work wants to study and find universal statistical laws in different kind of biological systems. If one finds that two different systems share some statistical laws and that they have a somehow similar data structure, therefore it is possible to use tools developed from different fields to study and gain information about each other. In particular two datasets containing information about some human healthy and diseased tissues will be analysed. These data come from biological experiments of RNA-sequencing.

The ultimate goal of this work would be to study, develop and build a machine learning's model which is able to discriminate healthy and diseased tissues at the beginning. Once diseased tissues are found, the next goal is to separate cancer types and ultimately sub-types, which is not always easy clinically. The interest in the development of a method able to separate well this kind of data is increasing~\cite{Farver2018}.

The methods to gain this goal are derived firstly from linguistics; in particular, a topic model approach will be widely described.

In chapter~\ref{ch:data}, I will describe the datasets used and introduce some basic biological properties of these datasets. In particular, I'll use two datasets of gene expression data from cancer diseases and healthy tissues.

In chapter~\ref{ch:structure}, I will describe the basics of component systems and give some useful mathematical definitions. Here it will be shown that RNA-sequencing data have many aspects in common with linguistics data. Examples of Zipf's law, well-known and in-depth studied in linguistics, will demonstrate that different sources of data (genomic and linguistics) can share some statistical properties. Some analyses are shown to explain the different behaviour of different tissues.

In chapter~\ref{ch:scalinglaws}, I will study the gene expression across samples of all the genes. This analysis is preparatory to the following sections where some gene selection would be necessary.

Demonstrated that linguistics and biological data share some statistical laws, in chapter~\ref{ch:topicmodelling}, the main one, I will describe how topic modelling can perform network analysis on these datasets. Topic modelling is an advanced clustering algorithm developed in linguistics to classify text and used in different fields of science. Different approaches to topic modelling are possible starting from the standard ones~\cite{Zhou2016} to some new proposal~\cite{Lancichinetti2015,Martini2017,gerlach2018network}. Using topic modelling one would find the inner structure of the data. One would find clusters such that all samples in a cluster share the tissue or the tumour type. Benchmarks and metrics to test and evaluate this algorithm will be widely discussed.

In chapter~\ref{ch:conclusions}, I will sum up the results and propose some future developments of this work.

Many methods of the pipeline, written in C\texttt{++} using openMP and Boost~\cite{siek2002boost}, are encapsulated in a tool available at~\url{https://github.com/fvalle1/tacos}. During this work I used different python libraries such as pandas~\cite{mckinney2010data}, scipy~\cite{jones2014scipy}, numpy~\cite{oliphant2006guide} and matplotlib~\cite{hunter2007matplotlib}. Some advanced analysis required Tensorflow~\cite{tensorflow2015-whitepaper} and pySpark~\cite{Zaharia:2016:ASU:3013530.2934664}. The topic modelling stochastic block model's minimization functions are implemented in the graph-tool library~\cite{peixoto_graph-tool_2014}. Computing resources was made available by EGI Foundation~\cite{fernandez2015egi} and from C$^{\text{3}}$S~\cite{occamchep}.

The full work repository is available on GitHub\textsuperscript{\tiny\textcopyright} at~\url{https://github.com/fvalle1/master_thesis} and runnable as Docker\textsuperscript{\tiny\textcopyright}  container pulling from~\url{https://cloud.docker.com/repository/docker/fvalle01/thesis}. 