\chapter*{Abstract}
The interest in studying complex systems is increasingly spreading.
Complex systems can be found anywhere and many common behaviours are
observable, systems with different origins and purposes may share, for
instance, some statistical laws.

An example can be the Zipf's law, well-known in linguistics and texts
analysis. It can be easily observed in the distribution of gene expressions
in different samples of cancer tissues.

In recent years datasets with a large amount of cancer samples' data are
available, the most complete is The Cancer Genome Atlas (TCGA). From
this dataset, it is easy to get, for example, gene expression data from
RNA-sequencing experiments together with a lot of information about the
samples themselves.

If one studies the number of samples in which a gene is expressed above
a certain threshold, the so-called occurrence, it is easily verified
that there are different kinds of genes. Some are present in the
majority of samples, some others are present only in a subset of the
whole dataset. The same behaviour can be found analysing words in
a corpus of texts; some words, such as \emph{the}, are present everywhere,
other specific words are present only in texts regarding a certain
subject. This suggests that there are similarities between a system of
words and documents and a system of genes and samples.

Given a corpus of documents, they can be classified by their specific
subject. Similarly, a set of samples can be classified, for
example, by the tissue it comes from or by the type of the disease it is
referred to.

The similarities between gene expression data and linguistics suggest
the possibility to use topic modelling to classify data and separate
samples and genes in different clusters. Topic modelling is a set of
clustering algorithms in networks' theory. Given a set of words and
documents, it describes documents as a mixture of topics. Topics are
nothing but communities of words each one with a given probability.

Purpose of this work is to build a bipartite network of genes and
samples and use topic modelling to find communities. The goal is to
separate samples depending on the site the tumour was and the disease
type of the sample. Moreover, it is possible to separate genes depending
on their specific functions. Once a community structure of genes
emerges, it is possible to run a hypergeometric test on the whole set to verify if they reveal some type of enrichment and to inspect
their common properties.

The specific algorithm used in this work is particularly unique because
it needs no priors and makes no assumptions on the data; moreover, it can
be set to accept overlapping clusters so it is possible to find genes
belonging to different topics and can be hierarchic all these facts
empower a lot of new possibilities to investigate the network.

A hierarchic approach make it possible to classify data at different
layers. An ideal goal would be to separate healthy and diseased samples
at the first layer, then separate by tissue, then by tumour type and so
on.
