\subsection{Run on Gene Tissues Expression dataset}
Once the model was set and adapted to RNA-Sequencing data, it was run on a subset of the GTEx dataset. A subset of samples was chosen randomly to reduce the computing time needed. The analysis hereby described took about 2 days to be run on a 16 core CPU, 100GB memory facility. The great amount of memory is needed to temporary store the network configuration at each step of the Monte Carlo simulation.

First of all, to rapidly have information about the interest of the oncoming result the metric above described were considered. In figure~\ref{fig:topic/gtex/oversigma_10tissue/metric_scores} it is represented the V-measure score versus the number of clusters found at different layers.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/metric_scores.pdf}
    \caption{Scores across the hierarchy. The performances classifying the primary site and secondary site are compared. Note that with one cluster the completeness is $1$ but the homogeneity is $0$ so the score goes down.}
    \label{fig:topic/gtex/oversigma_10tissue/metric_scores}
\end{figure}
The result is quite good, the maximum score is over $0.8$. Considering that, for example,~\cite{Farver2018} obtained a similar score analysing similar dataset considering just homogeneity, this can be considered a quite good result. A second interesting fact is that both the tissue label (primary site) and the sub-tissue label (secondary site) obtain such a good score. Moreover, the secondary site score's peak is at a higher number of clusters coherently with the fact that there is a greater number of sub-tissue labels. This score can be useful to extract the correct level of the hierarchy the consequent analysis should be made on.

In figure~\ref{fig:topic/gtex/oversigma_10tissue/bipartite_rebuild} the relationship between the clusters at different layers is evident. Each row is a layer of the hierarchy and arrows represent the path of a node across the hierarchy. Note that clusters don't overlap and the separation done at the greater level is maintained across all the hierarchy. This representation gives an idea of the point of the hierarchy where the tissues separate, see in figure~\ref{fig:topic/gtex/oversigma_10tissue/bipartite_rebuild} cluster $2$ that splits in two clusters separating \textit{breast} and \textit{adipose tissue}.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.8\linewidth]{pictures/topic/gtex/oversigma_10tissue/bipartite_rebuild.pdf}
    \caption{Hierarchy of the files' nodes. In the top the layers where the output consists of many small clusters, in the bottom the output with few big clusters. Arrows represent how nodes pass from a hierarchy layer to another. The bigger the balls, the bigger the cluster. The more yellow the link the more nodes are in common between the clusters of the two different layers. Plotted using clustree~\cite{clustree}.}
    \label{fig:topic/gtex/oversigma_10tissue/bipartite_rebuild}
\end{figure}

\clearpage
In figure~\ref{fig:topic/gtex/oversigma_10tissue/clustercomposition_l3_primary_site} each column is a cluster and each colour is a tissue of the dataset. It is evident that the majority of the tissues are identified: the first, second, fourth, fifth, sixth, seventh, and tenth columns are fully and uniformly coloured of the same colour. These correspond to an identification of \textit{brain}, \textit{skin}, \textit{lung}, \textit{blood} and \textit{testis}.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/clustercomposition_l3_primary_site.pdf}
    \caption{Clusters composition at the level of the hierarchy with the higher score. Each column is a cluster, each colour is a label.}
    \label{fig:topic/gtex/oversigma_10tissue/clustercomposition_l3_primary_site}
\end{figure}
In the normalized representation of the same clusters the homogeneity of the clusters is more evident.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/fraction_clustercomposition_l3_primary_site.pdf}
    \caption{Normalized composition of clusters. Again each column is a cluster, each colour is a label.}
    \label{fig:topic/gtex/oversigma_10tissue/fraction_clustercomposition_l3_primary_site}
\end{figure}
Going deeper in the hierarchy and looking at a layer with more cluster the result, shown in figure~\ref{fig:topic/gtex/oversigma_10tissue/fraction_clustercomposition_l2_primary_site}, demonstrates that, at this point, all the tissues are separated and each cluster is full of nodes sharing the same tissue.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/fraction_clustercomposition_l2_primary_site.pdf}
    \caption{Normalized composition of clusters at a deeper level.}
    \label{fig:topic/gtex/oversigma_10tissue/fraction_clustercomposition_l2_primary_site}
\end{figure}
Even looking at sub-tissues the results is quite good. It is not always easy to separate all the sub-parts of the \textit{brain}, nevertheless, the \textit{cerebellum} is well identified (column 27) and \textit{blood} is distinguished in \textit{whole blood} (columns 4-8) and \textit{lymphocytes} (column 3).
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/fraction_clustercomposition_l2_secondary_site.pdf}
    \caption{Normalized composition of clusters for the secondary site sub-tissue labels.}
    \label{fig:topic/gtex/oversigma_10tissue/fraction_clustercomposition_l2_secondary_site}
\end{figure}

\subsection{Null model shuffling labels}
A null model of cluster composition is necessary if one would be able to state that a result is better than expected. This was done by doing the same analysis but reshuffling the labels of the nodes. Reshuffling was done exchanging the label of each node with the one of another node picked up uniformly random. Doing so the number of clusters and the cluster sizes are maintained. In figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledclustercomposition_l3_primary_site} an example of clustering with random labels, it is evident that all clusters have similar and homogeneous composition. Note that not every tissue has the same number of samples, so, for example, \textit{brain} is more represented than other tissues.
\begin{figure}[htb!]
	\centering
	\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclustercomposition_l3_primary_site}
	\caption{Example of visualization of clusters with reshuffled labels.}
	\label{fig:topic/gtex/oversigma_10tissue/shuffledclustercomposition_l3_primary_site}
\end{figure}

All the results described in the previous pictures are quite qualitative. To have a more objective and mathematical measure of the success of the algorithm it is possible to measure the fraction of the most representative label in each cluster $k$
\[
max_{l\in labels}\left(\frac{n_{l k}}{n_k}\right)
\]
being $n_{l k}$ the number of nodes labelled $l$ in cluster $k$ and $n_k$ is the number of nodes in cluster $k$. This is represented in figure~\ref{fig:gtex/oversigma_10tissue/shuffledcluster_maximum_l2_primary_site} for the level where the V-measure is maximized (the best results are expected at this resolution). In figure~\ref{fig:gtex/oversigma_10tissue/shuffledcluster_maximum_l2_primary_site} on the left is shown the most representative label fraction for each cluster, on the right the histograms of the same quantity. Models' clusters are very homogeneous with the majority of cluster full, almost $100\%$, of the same tissue. It is also clear that reshuffling the labels the result is very different and so it is possible to admit that the models behave better than expected. 
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_maximum_l2_primary_site.pdf}
    \caption{The fraction of cluster composed by the representative label versus cluster. On the right the distributions of this measure.}
    \label{fig:gtex/oversigma_10tissue/shuffledcluster_maximum_l2_primary_site}
\end{figure}
In figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledcluster_maximum*} the same analysis is done for every level of the hierarchy. It is interesting to notice that at deeper levels (upper left in the figure) the random reshuffling and the real labels have the same behaviour. This happens because, at this level, clusters are very small and so it is easier to pick up nodes with the same label (in the extreme case of a cluster with size 1 it is always full of the same label). This shows that the deepest level is not interesting: results are the same with random labels; moreover the reshuffling null model is good to show up eventual biases due to small cluster sizes.
\begin{figure}[htb!]
    \centering
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_maximum_l0_primary_site.pdf}
    \end{minipage}
    \hspace{3mm}
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_maximum_l1_primary_site.pdf}
    \end{minipage}
    \\
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_maximum_l2_primary_site.pdf}
    \end{minipage}
    \hspace{3mm}
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_maximum_l3_primary_site.pdf}
    \end{minipage}
    \caption{The fraction of the most representative label in all clusters for different levels of the hierarchy. From upper left the deeper layer than downright the superficial one.}
    \label{fig:topic/gtex/oversigma_10tissue/shuffledcluster_maximum*}
\end{figure}
\FloatBarrier
A similar analysis can be made considering not just the number of the cluster but the cluster size, this is shown in figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l3_primary_site}. It is interesting to notice that the shuffle null model and the real labels clusters are different, so there must be some kind of signal. Clearly the model is able to output even big clusters full of the same label (points upper right in figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l3_primary_site}).
\begin{figure}[htb!]
	\centering
	\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l3_primary_site.pdf}
	\caption{The fraction of the most representative label versus cluster size.}
	\label{fig:topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l3_primary_site}
\end{figure}
In figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l*} the same analysis for all the levels of the hierarchy. It is interesting to see how going up in the hierarchy the two signals become different, as shown before the deeper layer (upper left in the image) is not different from null model and so it is not interesting.
\begin{figure}[htb!]
	\centering
	\begin{minipage}{0.45\textwidth}
		\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l0_primary_site.pdf}
	\end{minipage}
	\hspace{3mm}
	\begin{minipage}{0.45\textwidth}
		\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l1_primary_site.pdf}
	\end{minipage}
	\\
	\begin{minipage}{0.45\textwidth}
		\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l2_primary_site.pdf}
	\end{minipage}
	\hspace{3mm}
	\begin{minipage}{0.45\textwidth}
		\includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l3_primary_site.pdf}
	\end{minipage}
	\caption{The fraction of the most representative label versus cluster size across the hierarchy. From upper left the deeper layer than downright the superficial one.}
	\label{fig:topic/gtex/oversigma_10tissue/shuffledclusterhomosize_l*}
\end{figure}
\FloatBarrier
At this point to investigate deeper the structure of the clusters, it can be interesting to study how many labels are present in each cluster. The fraction of the most represented label defined above carries no information of what happens to the remaining labels. For example, if one cluster is composed of $80\%$ by label \textbf{A} and $20\%$ by label \textbf{B} and another cluster is composed $80\%$ by label \textbf{A}, $10\%$ by label \textbf{B} and $10\%$ by label \textbf{C} they have both a fraction of maximum representative label $80\%$ but the second in this example is more heterogeneous. Counting the number of different labels in each cluster can reveal this kind of effects. In figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l3_primary_site} it is represented the number of different labels versus cluster size. It is evident that the reshuffling case is quite different from the real one, almost every cluster in the null model has got every label. It is interesting to notice that the model outputs even big cluster with one label.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l3_primary_site.pdf}
    \caption{The number of different labels in each cluster versus cluster size.}
    \label{fig:topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l3_primary_site}
\end{figure}
In figure~\ref{fig:topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_lall} the same analysis for all the layer of the hierarchy. Even here the deeper level does not differ from the null model. Nevertheless, in layers with higher V-measure score, there is a strong signal that the reshuffling model is quite different from the model's output.
\begin{figure}[htb!]
    \centering
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l0_primary_site.pdf}
    \end{minipage}
    \hspace{3mm}
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l1_primary_site.pdf}
    \end{minipage}
    \\
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l2_primary_site.pdf}
    \end{minipage}
    \hspace{3mm}
    \begin{minipage}{0.45\textwidth}
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_l3_primary_site.pdf}
    \end{minipage}
\caption{The number of different labels in each cluster versus cluster size. From upper left the deeper layer than downright the superficial one.}
\label{fig:topic/gtex/oversigma_10tissue/shuffledcluster_shuffle_label_size_lall}
\end{figure}
\FloatBarrier
Having constructed the null model it is possible to estimate the V-measure score also for the null model. The results are reported in figure~\ref{fig:topic/gtex/oversigma_10tissue/metric_scores_shuffle}. 
Moreover, remembering the V-measure or normalized mutual information defined in~\ref{eq:mutualinformation} it is possible to estimate a mixed score which considers the homogeneity of the primary site and the completeness of the secondary site, doing so the score goes up if going deeper in the hierarchy the model makes more cluster with the same tissue but separates sub-tissues. It is not a big deal if one loses completeness regarding tissues (the model separates one big cluster full of the same label into two small ones) but gain information at the next resolution. This becomes clear if one looks at the big \textit{blood} cluster that in the next level of the hierarchy is separated into two clusters of \textit{blood}, one of \textit{whole blood} and one of \textit{lymphocytes}. The result is that this mixed score is the highest one.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/metric_scores_shuffle.pdf}
    \caption{Scores across the hierarchy. The scored is compared with some random labels. In blue the score for the primary site labels, in red for the secondary site labels, in yellow the shuffled labels, in green the mixed score with primary homogeneity and secondary completeness.}
    \label{fig:topic/gtex/oversigma_10tissue/metric_scores_shuffle}
\end{figure}
\FloatBarrier

\subsection{Comparison with standard algorithms}
At this point, it was verified that the model has got interesting output: it reaches high scores and has got a strong signal against the null model, at least at some levels of the hierarchy. It is now interesting to compare it with standard and well-studied similar algorithms.
First of all, a comparison is made with hierarchical clustering. This is done using the standard scipy package~\cite{jones2014scipy}, the metrics used was the euclidean one and the linkage method was set to Ward as briefly introduced in section~\ref{sec:hc}. This is quite fast, it needs a couple of minutes on a dual-core, 8GB memory machine.
In figure~\ref{fig:topic/gtex/oversigma_10tissue/metric_scores_hier} the comparison between these scores, the hierarchical algorithm performs worse than hierarchical Stochastic Block Model and as highly expected better than the random model.
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/metric_scores_hier.pdf}
    \caption{Scores across the hierarchy. The score obtained with hSBM is compared with hierarchical clustering and shuffled labels.}
    \label{fig:topic/gtex/oversigma_10tissue/metric_scores_hier}
\end{figure}

Another very used and well-studied algorithm is Latent Dirichlet Allocation briefly described in~\ref{sec:lda}. Running LDA, as implemented standard scipy package, is quite fast and is comparable with hierarchical clustering in terms of CPU time. Note that once LDA package extracts the topics, it is necessary to define some clusters, to do so a standard Agglomerative clustering approach was used, the distance was set to \textit{euclidean} and the linkage to \textit{Ward}. In figure~\ref{fig:topic/gtex/oversigma_10tissue/metric_scores_all} the V-measure scores for all the algorithms described until this point are reported. It is clear that the hierarchical Stochastic Block Model performs better than all the others, LDA gains a little worse score and hierarchical clustering is the worst of the three. It is highly expected that all models are quite different from the random model. The fact that hSBM and LDA have higher scores suggests that a topic model approach can be very useful in this kind of problems. 
\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\linewidth]{pictures/topic/gtex/oversigma_10tissue/metric_scores_all.pdf}
    \caption{Scores across hierarchy for all algorithms used in this work.}
    \label{fig:topic/gtex/oversigma_10tissue/metric_scores_all}
\end{figure}
Note that LDA and hierarchical cluster models were not fine-tuned and default parameters were used. Maybe a fine-tuning of these packages can lead to better and more satisfying results. This analysis, considering that the comparison was made with hierarchical Stochastic Block Model which is non-parametric and needs no setting, was done without any fine-tuning and standard parameters were set. This fact reveals another good point of hSBM, it extracts not only better clusters, but also the parameters necessary to this kind of models. Moreover, the number of topics was set to the one obtained from hSBM; LDA is not able to select the number of topics.

\FloatBarrier
\subsection{Topics enrichment tests}
The analyses up to this point considered only one of the two sides of the bipartite network; nothing was told yet about the genes (words in topic models).
The model outputs some clusters of genes, though.  From now on these clusters of genes will be called topics.

If one has got a set of genes, it is possible to perform an enrichment test to catch any important information and discover if there are any biological meanings behind it. Enrichment analysis checks whether an input set of genes significantly overlaps with annotated gene sets.
In this work tests were made using Gene Set Enrichment Analysis (GSEA)~\cite{subramanian2005gene} python tool~\cite{Kuleshov2016}, which performs a Fisher exact test (hypergeometric test). The Benjamini-Hochberg adjusted P-values is reported. Genes' annotation terms were searched in the following sets: 
\begin{itemize}
		\item GO\footnote{Gene Ontology} Molecular Function 2018
		\item GO Biological Process 2018
		\item GO Cellular Component 2018
		\item Human Phenotype Ontology
		\item Tissue Protein Expression from Human Proteome Map
		\item KEGG 2019 Human
		\item NCI-60 Cancer Cell Lines
		\item GTEx Tissue Sample Gene Expression Profiles up
		\item GTEx Tissue Sample Gene Expression Profiles down,
\end{itemize}
in particular the two latter contain annotation specific for GTEx dataset~\cite{Ardlie2015}.

In tables~\ref{tab:topic/enrich/pancreas},~\ref{tab:topic/enrich/brain} and~\ref{tab:topic/enrich/blood} examples of enrichment test results. Each table is a topic by hSBM. On the results it is put a P-value cut at $0.05$ and terms are sorted by the adjusted P-value.
\begin{table}[htb!]
	\tiny
	\begin{center}
		\begin{tabular}{|l|c|r|}
			\hline
			Term & \multicolumn{1}{l|}{Adjusted P-value} & Gene set \\ \hline
			pancreas male 60-69 years & 1E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 40-49 years & 3E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 40-49 years & 5E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 30-39 years & 1E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 20-29 years & 1E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 1E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 30-39 years & 1E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 2E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 40-49 years & 2E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 30-39 years & 2E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 2E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 20-29 years & 2E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 40-49 years & 3E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 50-59 years & 4E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 4E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 4E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 60-69 years & 5E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas female 50-59 years & 5E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 50-59 years & 5E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			pancreas male 30-39 years & 6E-18 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		\end{tabular}
	\end{center}
	\caption{Enrichment test of a topic. It is clear the enrichment for pancreas-related gene sets.}
	\label{tab:topic/enrich/pancreas}
\end{table}
\begin{table}[htb!]
	\tiny
	\begin{center}
		\begin{tabular}{|l|c|r|}
			\hline
			Term & \multicolumn{1}{l|}{Adjusted P-value} & Gene set \\ \hline
			brain female 40-49 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 50-59 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 40-49 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 40-49 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 60-69 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 50-59 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 50-59 years & 6E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 60-69 years & 7E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 50-59 years & 7E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 20-29 years & 7E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 8E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 8E-05 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 1E-04 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 1E-04 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain female 60-69 years & 1E-04 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
			brain male 60-69 years & 1E-04 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		\end{tabular}
	\end{center}
	\caption{Enrichment test of a topic. It is clear the enrichment for brain-related gene sets.}
	\label{tab:topic/enrich/brain}
\end{table}
\begin{table}[htb!]
	\centering
	\tiny
	\begin{tabular}{|l|c|r|}
		\hline
		Term & \multicolumn{1}{l|}{Adjusted P-value} & Gene set \\ \hline
		blood male 50-59 years & 3E-23 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 50-59 years & 3E-23 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 40-49 years & 3E-21 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 60-69 years & 9E-21 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 40-49 years & 3E-20 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 60-69 years & 4E-20 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 60-69 years & 4E-20 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 50-59 years & 5E-20 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 50-59 years & 1E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 60-69 years & 1E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 60-69 years & 1E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 60-69 years & 1E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 60-69 years & 2E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 50-59 years & 2E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 40-49 years & 2E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 40-49 years & 2E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 60-69 years & 2E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood male 30-39 years & 3E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 50-59 years & 5E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
		blood female 60-69 years & 5E-19 & GTEx Tissue Sample Gene Expression Profiles up \\ \hline
	\end{tabular}
	\caption{Enrichment test of a topic. It is clear the enrichment for blood-related gene sets.}
		\label{tab:topic/enrich/blood}
\end{table}
Tests were made on the topics at the level of the hierarchy which obtained the higher V-measure score on the sample side of the network. These results are very interesting, these enrichment tests demonstrate that not only the sample side of the network is well clustered but also the topics have a non-trivial meaning.

So also the topics are related to the tissues and somehow are tissue-specific. In the next examples, the relationship between the topics and the samples will be further investigated. In particular following what was done by~\cite{dey2017visualizing} the importance of each topic inside each sample, the $P(\text{topic} | \text{sample})$, will be discussed.

\FloatBarrier
Separate healthy tissues is a good exercise and a good benchmark for models, but the real goals would be being able to classify diseased samples. It is not always easy to identify and classify cancer tissues. In particular, being able to separate tumour sub-types would be the ideal pursuance of this work. So let's switch to the analysis of diseased samples.